deployer/commands/exec/infra_components.py

```diff
@@ -140,6 +200,17 @@ def root_homes(
         tmpf.flush()
 
         with cluster.auth():
+            # If we have an volume to restore from, create PV and PVC first
+            if restore_volume_id:
+                with tempfile.NamedTemporaryFile(mode="w", suffix=".json") as pv_tmpf:
```

Comment:
This is being created outside the `try` whose `finally` block contains the cleanup code, so it's possible that errors here will lead to the PV not being cleaned up. I think this hsould all be moved inside the same `try` whose `finally` has the cleanup code

Reply:
Ah, good catch, thank you!